File: backend/Wms.Api/Domain/Entities/Shipment.cs
namespace Wms.Api.Domain.Entities;

public class Shipment
{
    public int Id { get; set; }
    public required string ShipmentNumber { get; set; }
    public required string CarrierName { get; set; }
    public int OrderId { get; set; }
    public int LocationId { get; set; }
    public DateTime ShippedAtUtc { get; set; } = DateTime.UtcNow;
    public Order Order { get; set; } = null!;
    public Location Location { get; set; } = null!;
    public ICollection<ShipmentLine> Lines { get; set; } = [];
}

File: frontend/wms-web/src/types/shipment.ts
export type ShipmentLine = {
  id: number
  productId: number
  productSku: string
  productName: string
  quantity: number
}

export type Shipment = {
  id: number
  shipmentNumber: string
  carrierName: string
  orderId: number
  orderNumber: string
  customerName: string
  warehouseId: number
  warehouseName: string
  locationId: number
  locationCode: string
  shippedAtUtc: string
  lines: ShipmentLine[]
}

export type CreateShipmentRequest = {
  shipmentNumber: string
  carrierName: string
  orderId: number
  locationId: number
}
